src/binder/activitytracker.cpp

```diff
@@ -0,0 +1,80 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+// ============================================================
+//
+// activitytracker.cpp
+//
+
+
+//
+// Helpers for interaction with the managed ActivityTracker
+//
+// ============================================================
+
+#include "common.h"
+#include "activitytracker.h"
+
+void ActivityTracker::Start(const WCHAR *providerName, const WCHAR *activityName, /*out*/ GUID *activityId, /*out*/ GUID *relatedActivityId)
+{
+    GCX_COOP();
+    struct _gc {
+        OBJECTREF activityTracker;
+        STRINGREF providerName;
+        STRINGREF activityName;
+    } gc;
+    ZeroMemory(&gc, sizeof(gc));
+
+    GCPROTECT_BEGIN(gc);
+
+    MethodDescCallSite getInstance(METHOD__ACTIVITY_TRACKER__GET_INSTANCE);
+    gc.activityTracker = getInstance.Call_RetOBJECTREF(NULL);
+    gc.providerName = StringObject::NewString(providerName);
+    gc.activityName = StringObject::NewString(activityName);
+
+    MethodDescCallSite onStart(METHOD__ACTIVITY_TRACKER__ON_START, &gc.activityTracker);
```

Comment:
MethodDescCallSite is the older slower way of calling back to managed code. `CALL_MANAGED_METHOD` is the newer faster way of doing the same. Both of them work fine, but we tend to prefer to use the newer faster way for new code.

Reply:
Good to know - switched.